scanner/src/main/kotlin/Scanner.kt

```diff
@@ -587,31 +587,46 @@ class Scanner(
             }
         }
 
-        return try {
-            scanners.associateWith { scanner ->
-                logger.info { "Scan of $provenance with path scanner '${scanner.name}' started." }
+        val results = scanners.associateWith { scanner ->
+            logger.info { "Scan of $provenance with path scanner '${scanner.name}' started." }
 
-                // Filter the scan context to hide the excludes from scanner with scan matcher.
-                val filteredContext = if (scanner.matcher == null) context else context.copy(excludes = null)
-                val summary = scanner.scanPath(downloadDir, filteredContext)
+            // Filter the scan context to hide the excludes from scanner with scan matcher.
+            val filteredContext = if (scanner.matcher == null) context else context.copy(excludes = null)
 
-                logger.info { "Scan of $provenance with path scanner '${scanner.name}' finished." }
-
-                val summaryWithMappedLicenses = summary.copy(
-                    licenseFindings = summary.licenseFindings.mapTo(mutableSetOf()) {
-                        val licenseString = it.license.toString()
-                        it.copy(
-                            license = licenseString.mapLicense(scannerConfig.detectedLicenseMapping).toSpdx(),
-                            location = it.location.withRelativePath(downloadDir)
-                        )
-                    }
+            val summary = runCatching {
+                scanner.scanPath(downloadDir, filteredContext)
```

Comment:
commit-message: Could you describe, why catching `scanPath()` throws is now necessary and maybe outline a real world case where it throws?

Reply:
> why catching scanPath() throws is now necessary

It's not *now* necessary, but always was for proper handling of errors. I've clarified this in the commit message.